### PR TITLE
New Data Model for the Account Controller Create Method

### PR DESCRIPTION
**Pull request by kddejong**
## Proposed changes

- Update create account controller to use the new data model

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (changes to code, which do not change application behavior)

## Checklist

- [ ] I have filled out this PR template
- [ ] I have read the [CONTRIBUTING](https://github.com/Optum/dce/blob/master/docs/CONTRIBUTING.md) doc
- [ ] I have added automated tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (`README.md`, inline comments, etc.)
- [ ] I have updated the `CHANGELOG.md` under a `## next` release, with a short summary of my changes

### CODE REVIEW

File: cmd/lambda/accounts/create_test.go [hunk 1]
```diff
@@ -1,686 +1,95 @@
...
-			}),
-		).Return(&rolemanager.CreateRoleWithPolicyOutput{}, nil)
-
-		// Call the controller with the account
-		_, err := Handler(
-			context.TODO(),
-			createAccountAPIRequest(t, CreateRequest{
-				ID:           "1234567890",
-				AdminRoleArn: "arn:mock",
+			name: "success",
```
Comment (AmudaPalani) [inline cmd/lambda/accounts/create_test.go]: Can you please update the description to Gherkins style? I updated in lease to "When lease service - Get returns success".

File: cmd/lambda/accounts/main_test.go [hunk 2]
```diff
@@ -1,11 +1,9 @@
 package main
 
 import (
-	"encoding/json"
 	"os"
 	"testing"
 
-	"github.com/Optum/dce/pkg/api/response"
 	"github.com/aws/aws-lambda-go/events"
 )
```
Comment (AmudaPalani) [inline cmd/lambda/accounts/main_test.go]: Can you please add tests to run through the main's handler and init methods to call the create?

File: pkg/account/model.go [hunk 3]
```diff
@@ -1,22 +1,50 @@
...
+	StatusNone,
+	StatusLeased,
+	StatusNotReady,
+	StatusOrphaned,
+	StatusReady,
+}
+
+func init() {
+	var ok bool
+	PrincipalPolicyName, ok = os.LookupEnv("PRINCIPAL_POLICY_NAME")
```
Comment (joshmarsh) [inline pkg/account/model.go]: Why not inject this?

Comment (kddejong) [reply pkg/account/model.go]: That's not easily injectable.
1. As opposed to the Principal Role Name we don't store the policy name in the database.
2. When unmarshaling JSON/DynamoDB I want to populate this value so we can pass it along with the account.  That means I need it outside the struct and I need it some place those functions can get at it.

File: pkg/account/service.go [hunk 4]
```diff
@@ -133,6 +138,68 @@ func (a *Service) Update(ID string, data *Account) (*Account, error) {
...
+		ID:                *data.ID,
+		AdminRoleArn:      *data.AdminRoleArn,
+		Metadata:          data.Metadata,
+		PrincipalRoleName: a.principalRoleName,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	err = a.managerSvc.MergePrincipalAccess(new)
```
Comment (joshmarsh) [inline pkg/account/service.go]: Should we update this to UpsertPrincipalAccess?

File: pkg/account/service.go [hunk 5]
```diff
@@ -47,19 +48,24 @@ type ReaderWriterDeleter interface {
...
 
 // Manager manages all the actions against an account
 type Manager interface {
-	Setup(adminRole string) error
+	ValidateAccess(role *arn.ARN) error
+	MergePrincipalAccess(account *Account) error
 }
 
 // Service is a type corresponding to a Account table record
 type Service struct {
```
Comment (joshmarsh) [inline pkg/account/service.go]: It seems like 'AccountService' would be more accurate.

File: cmd/lambda/accounts/main_test.go (hunk 2 above)
Comment (kddejong) [reply cmd/lambda/accounts/main_test.go]: Done

File: pkg/account/service.go (hunk 4 above)
Comment (kddejong) [reply pkg/account/service.go]: Done

**Pull request closed without merging**